pylops_mpi/basicoperators/SecondDerivative.py

```diff
@@ -0,0 +1,230 @@
+import numpy as np
+
+from typing import Callable, Union
+from pylops.utils.typing import DTypeLike, InputDimsLike
+from pylops.utils._internal import _value_or_sized_to_tuple
+
+from pylops_mpi import DistributedArray, MPILinearOperator, Partition
+from pylops_mpi.utils.decorators import reshaped
+
+
+class MPISecondDerivative(MPILinearOperator):
+    r"""MPI Second Derivative
+
+    Apply a second derivative using a three-point stencil finite-difference
+    approximation with :class:`pylops_mpi.DistributedArray`. The Second Derivative
+    is calculated along ``axis=0``.
+
+    Parameters
+    ----------
+    dims : :obj:`int` or :obj:`tuple`
+        Number of samples for each dimension.
+    sampling : :obj:`float`, optional
+        Sampling step :math:`\Delta x`.
+    kind : :obj:`str`, optional
+        Derivative kind (``forward``, ``centered``, or ``backward``).
+    edge : :obj:`bool`, optional
+        Use reduced order derivative at edges (``True``) or
+        ignore them (``False``). This is currently only available
+        for centered derivative.
+    dtype : :obj:`str`, optional
+        Type of elements in the input array.
+
+    Attributes
+    ----------
+    shape : :obj:`tuple`
+        Operator shape
+
+    Notes
+    -----
+    The MPISecondDerivative operator applies a second derivative to a :class:`pylops_mpi.DistributedArray`
+    using either a second-order forward, backward or a centered stencil.
```

Comment:
Add similar comment as the one you did for the FirstDerivative or refer to it for implementation details

Reply:
Added a similar comment for ghosting...